packages/peregrine/lib/talons/CartPage/ProductListing/useQuantity.js

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useFieldState, useFieldApi } from 'informed';
+import debounce from 'lodash.debounce';
+
+/**
+ *  Quantity Component talon.
+ *
+ * @param {Number}      props.initialValue the initial quantity value
+ * @param {min}         props.min the minimum allowed quantity value
+ * @param {Function}    props.onChange change handler to invoke when quantity value changes
+ */
+export const useQuantity = props => {
+    const { initialValue, min, onChange } = props;
+
+    const [prevQuantity, setPrevQuantity] = useState(initialValue);
+
+    const quantityFieldApi = useFieldApi('quantity');
+    const { value: quantity } = useFieldState('quantity');
+
+    const isIncrementDisabled = useMemo(() => !quantity, [quantity]);
+
+    // "min: 0" lets a user delete the value and enter a new one, but "1" is
+    // actually the minimum value we allow to be set through decrement button.
+    const isDecrementDisabled = useMemo(() => !quantity || quantity <= 1, [
+        quantity
+    ]);
+
+    const debouncedOnChange = useMemo(() => debounce(onChange, 100), [
+        onChange
+    ]);
+
+    const handleDecrement = useCallback(() => {
+        const newQuantity = quantity - 1;
+        quantityFieldApi.setValue(newQuantity);
+        debouncedOnChange(newQuantity);
```

Comment:
@jimbo was this actually breaking for you? This is all sync so spamming the button should still work and the debounced change should get the value from the last click anyways.

Reply:
Yes. Ideally, the buttons would not be disabled while the request is in flight—I'd be able to click the button a dozen times, and while it would be firing requests in the background every few seconds, none of them would prevent me from continuing to click the button.

I think Revanth's approach of throttling the update may work best, since it saves on both the leading edge and the trailing edge.

Reply:
The issue was that my debounce delay was only 100ms. I made it 1000ms but we can discuss this with UX to fine tune.